### PR TITLE
Add workspace loading test case

### PR DESCRIPTION
**Pull request by maciektr**
**Stack**:
- #508
- #506
- #502
- #500
- #490
- #489
- #465
- #446
- #447 ⬅
- #445

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: scarb/tests/e2e/support/workspace_builder.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    pub fn add_member(mut self, member: impl Into<String>) -> Self {
+        self.members.push(member.into());
+        self
+    }
+
+    pub fn package(mut self, package: ProjectBuilder) -> Self {
+        self.package = Some(package);
+        self
+    }
```
Comment (mkaput) [inline scarb/tests/e2e/support/workspace_builder.rs:25]: This seems like a kinda rare, single test, usage scenario. Can't we just use `extra(&str)` pattern that we have in ProjectBuilder? It should simplify much of this code as far as I see.

**Review by mkaput — CHANGES_REQUESTED** [review]

File: scarb/tests/e2e/support/workspace_builder.rs (hunk 1 above)
Comment (maciektr) [reply scarb/tests/e2e/support/workspace_builder.rs]: Hmm, I am not sure about this one. 
Root packages can be tricky sometimes, as their location in workspace root can cause some weird edge-cases (e.g. our offline discussion about fmt), so naturally we may want more than one test case. 
Also, eradicating this field would make us to create package sources, pass the package manifest, sometimes also concatenated with other manifest extras on our own, which may be inconvenient at times.

Comment (mkaput) [reply scarb/tests/e2e/support/workspace_builder.rs]: ok, convinced

**Review by mkaput — APPROVED** [review]

**Review by mkaput — APPROVED** [review]

**Comment by maciektr** [comment]
✓ Commit merged in pull request [#446](https://github.com/software-mansion/scarb/pull/446)
